src/timeline/timeline.style.tsx

```diff
@@ -1,60 +1,99 @@
 import styled, { css } from "styled-components";
+import { TickCircleFillIcon } from "@lifesg/react-icons";
 import { Color } from "../color";
 import { MediaQuery } from "../media";
 import { Text } from "../text";
-import { TimelineStatusProps } from "./types";
+import { TimelineStatusProps, Variant } from "./types";
 
 // =============================================================================
 // STYLE INTERFACES, transient props are denoted with $
 // See more https://styled-components.com/docs/api#transient-props
 // =============================================================================
 
-interface CircleIndicatorStyleProps {
-    $filled?: boolean | undefined;
-}
-
 interface TimelineWrapperStyleProps {
     $startCol?: number | undefined;
     $colSpan?: number | undefined;
 }
 
+interface VariantStyleProps {
+    $variant: Variant;
+}
+
 // =============================================================================
 // STYLE INTERFACES
 // =============================================================================
-
-export const CircleIndicator = styled.div<CircleIndicatorStyleProps>`
-    width: 1rem;
-    height: 1rem;
-    margin-top: 0.5rem;
+export const CircleIndicator = styled.div<VariantStyleProps>`
+    overflow: hidden;
+    width: 1.5rem;
+    height: 1.5rem;
+    margin-top: 4px;
     border-radius: 50%;
+
     ${(props) => {
-        if (props.$filled) {
-            return css`
-                border: none;
-                background-color: ${Color.Accent.Light[1]};
-            `;
+        switch (props.$variant) {
+            case "current":
+                return css`
+                    background-color: ${Color.Accent.Light[1]};
+                `;
+            case "upcoming-active":
+                return css`
+                    border: 4px solid ${Color.Accent.Light[1]};
+                `;
+            case "upcoming-inactive":
+                return css`
+                    border: 4px solid ${Color.Neutral[4]};
+                `;
+            case "completed":
+                return css`
+                    svg {
+                        display: block;
+                    }
+                `;
         }
-
-        return css`
-            border: 3.2px solid ${Color.Accent.Light[1]};
-            background-color: transparent;
-        `;
     }}
 `;
 
-export const LineIndicator = styled.div`
-    width: 0.25rem;
+export const TickCircleIcon = styled(TickCircleFillIcon)`
+    display: none;
+    width: 100%;
+    height: 100%;
+    transform: scale(1.2);
+
+    path {
+        fill: ${Color.Validation.Green.Icon};
+    }
+`;
```

Comment:
does `color` not work here?

Also I wonder if you can avoid the `transform` by using `TickIcon` instead. The circle indicator would still provide the green background, and the tick can be coloured white